--- a/nav2_navfn_planner/src/navfn.cpp
+++ b/nav2_navfn_planner/src/navfn.cpp
@@ -901,10 +901,10 @@ NavFn::calcPath(int n, int * st)
       dy += y * ss;
 
       // check for overflow
-      if (dx > 1.0) {stc++; dx -= 1.0;}
-      if (dx < -1.0) {stc--; dx += 1.0;}
-      if (dy > 1.0) {stc += nx; dy -= 1.0;}
-      if (dy < -1.0) {stc -= nx; dy += 1.0;}
+      if (dx > 0.5) {stc++; dx -= 1.0;}
+      if (dx < -0.5) {stc--; dx += 1.0;}
+      if (dy > 0.5) {stc += nx; dy -= 1.0;}
+      if (dy < -0.5) {stc -= nx; dy += 1.0;}
     }
 
     //      ROS_INFO("[Path] Pot: %0.1f  grad: %0.1f,%0.1f  pos: %0.1f,%0.1f\n",
